src/components/CharacterPage.js

```diff
@@ -1,23 +1,41 @@
 import React from "react";
 import { InfoField } from "./InfoField";
 import { Characteristics } from "./Characteristics";
+import { useQuery, gql } from "@apollo/client";
 import { sampleData } from "../mock-data";
 
+const CHARACTER = gql`
+  query getCharacter {
+    character(id: 1){
+      id
+      name
+      species
+      specialization
+      career
+    } 
+  }
+`
+
 export const CharacterPage = () => {
-  const character = sampleData.data.attributes.character;
+  const characterSamp = sampleData.data.attributes.character;
+  const { loading, error, data } = useQuery(CHARACTER);
+
+  if (loading) return 'Loading your data...';
+  if (error) return `Error! ${error.message}`;
+
   const characterInfo = (
     <>
-    <InfoField heading={Object.keys(character)[0]} info={character.name}/>
-    <InfoField heading={Object.keys(character)[1]} info={character.species}/>
-    <InfoField heading={Object.keys(character)[2]} info={character.career}/>
-    <InfoField heading={Object.keys(character)[3]} info={character.specialization}/>
+    <InfoField heading={Object.keys(data.character)[3]} info={data.character.name}/>
```

Comment:
I'm curious about this method of interpolating that data. Could these be accessed by using the actual key?

Reply:
Yeah, I don't feel like this is an optimal solution. I'd love to circle back to this and discuss once we have more data to work with!